src/creature.cpp

```diff
@@ -2833,6 +2835,26 @@ body_part_set Creature::get_drenching_body_parts( bool upper, bool mid, bool low
     return ret;
 }
 
+std::vector<bodypart_id> Creature::get_ground_contact_bodyparts( bool arms_legs ) const
+{
+    std::vector<bodypart_id> bodyparts;
+    if( has_effect( effect_quadruped_full ) || has_effect( effect_quadruped_half ) ) {
+        if( arms_legs == true ) {
+            bodyparts = { bodypart_id( "arm_l" ), bodypart_id( "arm_r" ), bodypart_id( "leg_l" ), bodypart_id( "leg_r" ) };
+        } else {
+            bodyparts = { bodypart_id( "hand_l" ), bodypart_id( "hand_r" ), bodypart_id( "foot_l" ), bodypart_id( "foot_r" ) };
+        }
+        return bodyparts;
+    } else {
+        if( arms_legs == true ) {
+            bodyparts = { bodypart_id( "leg_l" ), bodypart_id( "leg_r" ) };
+        } else {
+            bodyparts = { bodypart_id( "foot_l" ), bodypart_id( "foot_r" ) };
```

Comment:
Generate these lists with `get_body_parts_of_type` instead of hardcoding in the ids.